### PR TITLE
Datasource: Use Card component for the DataSourceList

### PR DESCRIPTION
**Pull request by Clarity-89**
**What this PR does / why we need it**:
Use the new Card component for the DataSourceList. 

Before: 
Screenshot 2021-02-18 at 16 47 23

After:
Screenshot 2021-02-18 at 16 47 35

Couldn't find the same tag color for `default` tag, so using primary color instead.

### CODE REVIEW

File: public/app/features/datasources/DataSourcesList.tsx [hunk 1]
```diff
@@ -1,39 +1,58 @@
...
-      'card-section': true,
-      'card-list-layout-grid': layoutMode === LayoutModes.Grid,
-      'card-list-layout-list': layoutMode === LayoutModes.List,
-    });
+export const DataSourcesList: FC<Props> = ({ dataSources, layoutMode }) => {
+  const listStyle = classNames({
+    'card-section': true,
+    'card-list-layout-grid': layoutMode === LayoutModes.Grid,
+    'card-list-layout-list': layoutMode === LayoutModes.List,
+  });
```
Comment (Clarity-89) [inline public/app/features/datasources/DataSourcesList.tsx:21]: Don't think this is needed as I haven't noticed any other components using `layoutMode`, but not 100% sure it's safe to delete.

Comment (torkelo) [reply public/app/features/datasources/DataSourcesList.tsx:21]: yea, you can delete these, not used

**Review by torkelo — APPROVED** [review]
Awesome! 

Found two issues (not solved) 

* No hover background effect 
* The logo for github data source is smaller so things do not align (not sure what's special about this logo)

Screenshot from 2021-02-19 10-57-37

Also fixed some issues
* Made heading bold
* Switched to type name instead of type id

**Comment by torkelo** [comment]
Screenshot from 2021-02-19 13-03-32

**Comment by torkelo** [comment]
@Clarity-89 what is this after element for? cannot see in dev tools and seems unnecessary? 
https://github.com/grafana/grafana/blob/master/packages/grafana-ui/src/components/Card/Card.tsx#L154

**Comment by Clarity-89** [comment]
> @Clarity-89 what is this after element for? cannot see in dev tools and seems unnecessary?
> https://github.com/grafana/grafana/blob/master/packages/grafana-ui/src/components/Card/Card.tsx#L154

This is an overlay in case the card is disabled, using ::after instead of creating a separate HTML element.

**Comment by Clarity-89** [comment]
> Awesome!
> 
> Found two issues (not solved)
> 
> * No hover background effect
> * The logo for github data source is smaller so things do not align (not sure what's special about this logo)
> 
> Screenshot from 2021-02-19 10-57-37
> 
> Also fixed some issues
> 
> * Made heading bold
> * Switched to type name instead of type id

Do we need to have `type` word there? An icon would be better but I guess it's hard to find a good one for this use case. Also this will make the heading bold in all the Cards, not sure if this will work for all the use cases?

File: packages/grafana-data/src/types/datasource.ts [hunk 2]
```diff
@@ -549,7 +549,7 @@ export interface DataSourceSettings<T extends DataSourceJsonData = DataSourceJso
   name: string;
   typeLogoUrl: string;
   type: string;
-  typeName: string;
+  typeName?: string;
```
Comment (torkelo) [inline packages/grafana-data/src/types/datasource.ts:552]: it's not optional

**Comment by torkelo** [comment]
> Do we need to have type word there? 

I think that was from a screenshot where I tested that prefix

File: packages/grafana-data/src/types/datasource.ts (hunk 2 above)
Comment (Clarity-89) [reply packages/grafana-data/src/types/datasource.ts:552]: Hmm, there are a lot of type errors in other components due to the missing `typeName`.

**Review by torkelo — APPROVED** [review]

File: public/app/features/datasources/DataSourcesList.tsx [hunk 3]
```diff
@@ -1,39 +1,50 @@
...
-        </ol>
-      </section>
-    );
-  }
-}
+  return (
+    <ul className={styles.list}>
+      {dataSources.map((dataSource, index) => {
+        return (
+          <li key={`${dataSource.id}-${index}`}>
```
Comment (joshhunt) [inline public/app/features/datasources/DataSourcesList.tsx:22]: fwiw, there's no value in concatenating these together - ``${dataSource.id}-${index}`` is essentially the same as `index` as the non-stable index will "poison" the whole key.

if `dataSource.id` is guarenteed to be unique and stable then just that'll do :)

Comment (Clarity-89) [reply public/app/features/datasources/DataSourcesList.tsx]: Good catch!

**Pull request merged**